### PR TITLE
AP-1733: Migrate assets to webpack

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-1733)

During learning time I'm trying to understand the differences between Sprockets and Webpacker
As part of that, I started to move the assets away from sprockets

This PR moves a lot of files from the legacy, `app/assets` folder and into the newly named `app/webpack` folder.
It redirects all of the load paths to use the new pack file locations and switches helpers and config appropriately too

- [x] Move images to webpacker
- [x] Move fonts to webpacker
- [x] Move css to webpacker
- [x] Make sure it's deployable!
- [x] remove sprockets config
- [x] remove sprockets gem
- [x] End-to-end regression test including PDF generation 

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ gem 'pg'
 gem 'puma', '~> 5.0'
 gem 'rails', '~> 6.0.3'
 gem 'regexp-examples'
-gem 'sass-rails', '~> 6.0', '>= 6.0.0'
+# gem 'sass-rails', '~> 6.0', '>= 6.0.0'
```
Comment (Obsiye) [inline Gemfile:17]: left over comment

File: app/webpack/stylesheets/check-your-answers.scss [hunk 2]
```diff
@@ -1,7 +1,7 @@
-@import "govuk-frontend/govuk/settings/all";
-@import "govuk-frontend/govuk/tools/all";
-@import "govuk-frontend/govuk/helpers/all";
-$govuk-fonts-path: "/assets/";
+@import "~govuk-frontend/govuk/settings/all";
+@import "~govuk-frontend/govuk/tools/all";
+@import "~govuk-frontend/govuk/helpers/all";
+//$govuk-fonts-path: "";
```
Comment (Obsiye) [inline app/webpack/stylesheets/check-your-answers.scss:4]: left over comment.

File: app/webpack/stylesheets/moj-banner.scss [hunk 3]
```diff
@@ -1,7 +1,7 @@
-@import "govuk-frontend/govuk/settings/all";
-@import "govuk-frontend/govuk/tools/all";
-@import "govuk-frontend/govuk/helpers/all";
-$govuk-fonts-path: "/assets/";
+@import "~govuk-frontend/govuk/settings/all";
+@import "~govuk-frontend/govuk/tools/all";
+@import "~govuk-frontend/govuk/helpers/all";
+//$govuk-fonts-path: "/assets/";
```
Comment (Obsiye) [inline app/webpack/stylesheets/moj-banner.scss:4]: left over comment

File: config/application.rb [hunk 4]
```diff
@@ -1,6 +1,19 @@
...
-require 'rails/all'
+require 'rails'
+# Pick the frameworks you want:
+require 'active_model/railtie'
+require 'active_job/railtie'
+require 'active_record/railtie'
+require 'active_storage/engine'
+require 'action_controller/railtie'
+require 'action_mailer/railtie'
+require 'action_mailbox/engine'
```
Comment (Obsiye) [inline config/application.rb:11]: do we need this 'action_mailbox/engine' or 'action_cable/engine'

**Review by mpw5 — APPROVED** [review]
I second Abdi's comments about left over comments, but apart from that it looks great to me. Awesome work 🎉

**Comment by colinbruce** [comment]
I had to rebase to resolve a merge conflict in Gemfile.lock but otherwise, just a commit to remove the comments

**Pull request merged**